### PR TITLE
Add timestamp into Jobdata for more informative display by commenter

### PR DESCRIPTION
**Pull request by Yizong98**

### CODE REVIEW

File: tools/flaky-test-retryer/prowapi/types.go [hunk 1]
```diff
@@ -117,4 +117,5 @@ type ReportMessage struct {
 	Refs    []Refs       `json:"refs,omitempty"`
 	JobType ProwJobType  `json:"job_type"`
 	JobName string       `json:"job_name"`
+	Timestamp int64      `json:"timestamp"`
```
Comment (chaodaiG) [inline tools/flaky-test-retryer/prowapi/types.go:120]: Are you sure this field exist in the pubsub message?

File: tools/flaky-test-retryer/handler.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ package main
 
 import (
 	"context"
+	"time"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:25]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/log_parser.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ import (
 	"log"
 	"path/filepath"
 	"strings"
+	"time"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser.go:27]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/log_parser.go [hunk 4]
```diff
@@ -47,6 +48,7 @@ func InitLogParser(serviceAccount string) error {
 // and a cached flaky report it is referencing.
 type JobData struct {
 	*prowapi.ReportMessage
+	timestamp time.time
```
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser.go:51]: Format Go code:
```suggestion
	timestamp    time.time
```

File: tools/flaky-test-retryer/subscriber/subscriber.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ import (
 	"context"
 	"encoding/json"
 	"log"
+	"time"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/subscriber/subscriber.go:23]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/subscriber/subscriber.go [hunk 6]
```diff
@@ -54,12 +55,12 @@ func NewSubscriberClient(subName string) (*Client, error) {
...
 // ReceiveMessageAckAll acknowledges all incoming pusub messages and convert the pubsub message to ReportMessage.
 // It executes `f` only if the pubsub message can be converted to ReportMessage. Otherwise, ignore the message.
-func (c *Client) ReceiveMessageAckAll(ctx context.Context, f func(*prowapi.ReportMessage)) error {
+func (c *Client) ReceiveMessageAckAll(ctx context.Context, f func(*prowapi.ReportMessage, time.time)) error {
 	return c.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
 		if rmsg, err := c.toReportMessage(msg); err != nil {
 			log.Printf("Cannot convert pubsub message (%v) to Report message %v", msg, err)
 		} else if rmsg != nil {
-			f(rmsg)
+			f(rmsg,msg.PublishTime)
```
Comment (mattmoor) [inline tools/flaky-test-retryer/subscriber/subscriber.go:63]: Format Go code:
```suggestion
			f(rmsg, msg.PublishTime)
```

File: tools/flaky-test-retryer/handler.go (hunk 2 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:25]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/log_parser.go [hunk 7]
```diff
@@ -47,6 +48,7 @@ func InitLogParser(serviceAccount string) error {
 // and a cached flaky report it is referencing.
 type JobData struct {
 	*prowapi.ReportMessage
+	Timestamp	time.Time
```
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser.go:51]: Format Go code:
```suggestion
	Timestamp    time.Time
```

File: tools/flaky-test-retryer/subscriber/subscriber.go [hunk 8]
```diff
@@ -54,12 +55,12 @@ func NewSubscriberClient(subName string) (*Client, error) {
...
 // ReceiveMessageAckAll acknowledges all incoming pusub messages and convert the pubsub message to ReportMessage.
 // It executes `f` only if the pubsub message can be converted to ReportMessage. Otherwise, ignore the message.
-func (c *Client) ReceiveMessageAckAll(ctx context.Context, f func(*prowapi.ReportMessage)) error {
+func (c *Client) ReceiveMessageAckAll(ctx context.Context, f func(*prowapi.ReportMessage, time.Time)) error {
 	return c.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
 		if rmsg, err := c.toReportMessage(msg); err != nil {
 			log.Printf("Cannot convert pubsub message (%v) to Report message %v", msg, err)
 		} else if rmsg != nil {
-			f(rmsg)
+			f(rmsg,msg.PublishTime)
```
Comment (mattmoor) [inline tools/flaky-test-retryer/subscriber/subscriber.go:63]: Format Go code:
```suggestion
			f(rmsg, msg.PublishTime)
```

File: tools/flaky-test-retryer/handler.go (hunk 2 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:25]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/handler.go [hunk 9]
```diff
@@ -22,6 +22,7 @@ package main
...
 	"fmt"
 	"log"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:27]: Format Go code:
```suggestion
	"log"
	"time"
```

File: tools/flaky-test-retryer/log_parser.go (hunk 7 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser.go:51]: Format Go code:
```suggestion
	Timestamp    time.Time
```

File: tools/flaky-test-retryer/subscriber/subscriber.go (hunk 8 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/subscriber/subscriber.go:63]: Format Go code:
```suggestion
			f(rmsg, msg.PublishTime)
```

File: tools/flaky-test-retryer/handler.go (hunk 2 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:25]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/handler.go (hunk 9 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:27]: Format Go code:
```suggestion
	"log"
	"time"
```

File: tools/flaky-test-retryer/handler.go (hunk 2 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:25]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/handler.go (hunk 9 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/handler.go:27]: Format Go code:
```suggestion
	"log"
	"time"
```

File: tools/flaky-test-retryer/log_parser_test.go [hunk 10]
```diff
@@ -101,9 +102,9 @@ func testIsSupported(t *testing.T) {
 				Repo:  fakeRepo,
 				Pulls: nil,
 			}},
-		}, nil, nil}, false},
-		{&JobData{fakeInvalidRepo, nil, nil}, false}, // invalid repo
-		{&JobData{fakeValidMessage, nil, nil}, true}, // valid message
+		}, time.Now(),nil, nil}, false},
```
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

File: tools/flaky-test-retryer/github_commenter.go [hunk 11]
```diff
@@ -26,6 +26,7 @@ import (
 	"sort"
 	"strconv"
 	"strings"
+	"time"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/github_commenter.go:29]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/log_parser_test.go (hunk 10 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

File: tools/flaky-test-retryer/github_commenter.go (hunk 11 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/github_commenter.go:29]: Format Go code:
```suggestion

```

File: tools/flaky-test-retryer/log_parser_test.go (hunk 10 above)
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser_test.go:105]: Format Go code:
```suggestion
		}, time.Now(), nil, nil}, false},
```

**Comment by chaodaiG** [comment]
Looks good

/lgtm
/approve

**Pull request merged**